beacon-chain/state/state-native/v1/setters_misc.go

```
@@ -54,76 +55,68 @@ func (b *BeaconState) SetGenesisValidatorRoot(val []byte) error {
 	b.lock.Lock()
 	defer b.lock.Unlock()
 
-	b.state.GenesisValidatorsRoot = val
+	b.genesisValidatorsRoot = bytesutil.ToBytes32(val)
```

we should return an error if `val` isnt 32 bytes long.